src/scalarstats.jl

```diff
@@ -271,6 +367,29 @@ end
 zscore{T<:Real}(X::AbstractArray{T}) = ((μ, σ) = mean_and_std(X); zscore(X, μ, σ))
 zscore{T<:Real}(X::AbstractArray{T}, dim::Int) = ((μ, σ) = mean_and_std(X, dim); zscore(X, μ, σ))
 
+"""
+    zscore(X::AbstractArray, μ::AbstractArray, σ::AbstractArray)
+
+### Args:
+* `X`: The input matrix to compute the Z-score.
+* `μ`: The mean, this could be a scalar or a an array.
+* `σ`: The standard deviation, this could be a scalar or a an array.
+
+Compute the Z-scores, given the mean μ and standard deviation σ, which is defined as (x - μ) / σ. This function returns an array Z of the same size as X. Here, μ and σ should be both scalars or both arrays.
+"""
+function zscore end
+
+"""
+    zscore(X::AbstractArray, μ::AbstractArray, σ::AbstractArray)
+
+### Args:
+* `X`: The input matrix to compute the Z-score.
+* `μ`: The mean, this could be a scalar or a an array.
+* `σ`: The standard deviation, this could be a scalar or a an array.
+
+Compute the Z-scores inline, given the mean μ and standard deviation σ, which is defined as (x - μ) / σ. Here, μ and σ should be both scalars or both arrays.
```

Comment:
"in-place" instead of "inline". Would likely be clearer by also saying "storing them directly in `X`".